src/Order/Base.php

```diff
@@ -711,7 +704,8 @@ public function put_label_content( $response, $order, $parent_barcode, $parent_l
 					$label_extension = ! empty( $label_contents['OutputType'] ) ? sanitize_title( $label_contents['OutputType'] ) : 'pdf';
 					$barcode         = $response[ $type ][ $shipment_idx ][ $content_type['barcode_key'] ];
 					$barcode         = is_array( $barcode ) ? array_shift( $barcode ) : $barcode;
-					$filename        = Utils::generate_label_name( $order->get_id(), $label_type, $barcode, 'A6', $label_extension );
+					$label_format  	 = $this->settings->get_label_format();
+					$filename        = Utils::generate_label_name( $order->get_id(), $label_type, $barcode, $label_format, $label_extension );
```

Comment:
but the label file returned from API is always **A6**, is there a reason for this?

Reply:
@Abdalsalaam It cause a fatal error when it generate files with format A4 then try to create the label again from bulk action because file path is A6 but the one saved to database is A4